plugins/net_plugin/net_plugin.cpp

```diff
@@ -4320,47 +4422,87 @@ namespace eosio {
    }
 
    void connections_manager::connect_supplied_peers(const string& p2p_address) {
-      std::lock_guard g(connections_mtx);
       for (const auto& peer : supplied_peers) {
-         connect_i(peer, p2p_address);
+         resolve_and_connect(peer, p2p_address);
       }
    }
 
    void connections_manager::add( connection_ptr c ) {
       std::lock_guard g( connections_mtx );
-      add_i( std::move(c) );
+      boost::system::error_code ec;
+      auto endpoint = c->socket->remote_endpoint(ec);
+      connections.insert( connection_details{
+         .host = c->peer_address(), 
+         .c = std::move(c),
+         .active_ip = endpoint} );
    }
 
    // called by API
    string connections_manager::connect( const string& host, const string& p2p_address ) {
+      supplied_peers.insert(host);
```

Comment:
`connect` can be called from multiple threads from the http thread pool at the same time. This needs to be protected by a mutex.